### PR TITLE
Renaming proj4 to proj.4

### PR DESCRIPTION
**Pull request by ocefpaf**
Just experimenting.  This PR:

- Removes `proj4 4.9.1`
- Change cartopy to use the default channel version of Proj4 (`proj.4 4.9.1`)

Note that the default channel does not provide version 4.9.2 of Proj4 yet.
Anyways, I am not sure if that version is OK for cartopy.  See https://github.com/ioos/conda-recipes/pull/500.

### CODE REVIEW

File: cartopy/meta.yaml [hunk 1]
```diff
@@ -10,8 +10,8 @@ source:
     - cartopy.win.patch  # [win]
 
 build:
-  number: 0  # [not linux]
-  number: 1  # [linux]
+  number: 1  # [not linux]
+  number: 2  # [linux]
```
Comment (QuLogic) [inline cartopy/meta.yaml]: Now that they're both bumped, can we not just make them the same?

Comment (ocefpaf) [inline cartopy/meta.yaml]: If we bump both to release 2 then the channel would not have release 1 for OSX and Windows.  It is a minor inconsistency (and I do that in the IOOS channel all the time), but that does not seems to be the way @pelson do things here.

**Comment by ocefpaf** [comment]
I am almost certain that the circleci failure is a conda bug that is already fixed in the latest version.  @pelson  do you want to add a `conda update --yes --all` to the `run_docker_build.sh` file or do you want to update the docker image?

File: cartopy/meta.yaml (hunk 1 above)
Comment (pelson) [inline cartopy/meta.yaml]: I'd be happy with that.

**Comment by pelson** [comment]
Notice the numpy build string isn't in the cartopy distribution:

```
	cartopy-0.13.0-py27_1 (will be built: True)
	cartopy-0.13.0-py27_1 (will be built: True)
	cartopy-0.13.0-py34_1 (will be built: True)
```

Would you mind adding the necessary ``??`` to the recipe?

**Comment by ocefpaf** [comment]
> Notice the numpy build string isn't in the cartopy distribution:

I thought that `numpy *` would take care of that :unamused: 

> Would you mind adding the necessary ?? to the recipe?

I will make a few tests and see if I can fix that.

File: proj.4/build.sh [hunk 2]
```diff
@@ -1,5 +1,11 @@
 #!/bin/bash
 
 ./configure --prefix=$PREFIX --without-jni
+
 make
+
+if [[ $(uname) == Linux ]]; then
+    make check
+fi
```
Comment (ocefpaf) [inline proj.4/build.sh]: NB `make check fails in OSX`! We did not run check before.

File: cartopy/meta.yaml (hunk 1 above)
Comment (ocefpaf) [inline cartopy/meta.yaml]: Done!

**Comment by ocefpaf** [comment]
I am trying a different approach here to get more control over the build string. I just noticed that it is gone from a few other packages too.  I am not sure if the disappearance of the build string was something I changed or a conda upgrade.

Right now change https://github.com/SciTools/conda-recipes-scitools/commit/fad5239c4bc2761aa307b25f2618abed3dd99355 produces `cartopy-0.13.0-npNonepy27_2`.

`None` would be replaced by the proper numpy build string if we have either the `CONDA_NPY=$VER` env variable or if we call the build with `conda build --numpy=$VER ...`.

@pelson does Obvious-CI creates the $VER variable for numpy?  Is this an approach you want to pursue?

If that works and you are OK with it I can propagate that change to the other packages.

**Comment by ocefpaf** [comment]
Actually that produced all these:

- `cartopy-0.13.0-npNonepy27_2`
- `cartopy-0.13.0-np18py27_2`
- `cartopy-0.13.0-np19py27_2`
- `cartopy-0.13.0-np110py27_2`

I guess we just need to find a way to get rid of `cartopy-0.13.0-npNonepy27_2`.

File: proj.4/build.sh (hunk 2 above)
Comment (QuLogic) [inline proj.4/build.sh]: For 4.9.2 as well? What is the error? We should push that up to proj.4.

Comment (ocefpaf) [inline proj.4/build.sh]: > For 4.9.2 as well?

Don't remember, but I had other issues building cartopy on Windows against Proj.4 4.9.2.

> What is the error?

https://travis-ci.org/ioos/conda-recipes/builds/84817022

> We should push that up to proj.4.

Not sure what you mean.

Comment (QuLogic) [inline proj.4/build.sh]: That build is failing with cdo, not proj.4.

Comment (ocefpaf) [inline proj.4/build.sh]: Yep.  I mixed the logs here.  Let me see if I can find it.  (Or better yet, remove the conditional and see if it fails again.)

Comment (ocefpaf) [inline proj.4/build.sh]: @QuLogic 

Here is Proj 4.9.2 failure: https://travis-ci.org/ioos/conda-recipes/builds/85402318

And as soon as travis finishes we will know what happens here with proj 4.9.1

Comment (ocefpaf) [inline proj.4/build.sh]: Same error: https://travis-ci.org/SciTools/conda-recipes-scitools/builds/85403903

I am removing that `make check` again.

Comment (QuLogic) [inline proj.4/build.sh]: Can you try adding this patch?
```patch
diff --git a/src/rtodms.c b/src/rtodms.c
index 591874a..96231ff 100644
--- a/src/rtodms.c
+++ b/src/rtodms.c
@@ -63,7 +63,7 @@ rtodms(char *s, double r, int pos, int neg) {
                if (*p != '.')
                        ++p;
                if (++q != p)
-                       (void)strcpy(p, q);
+                       (void)memmove(p, q, (sign ? 3 : 2));
        } else if (min)
                (void)sprintf(ss,"%dd%d'%c",deg,min,sign);
        else
```

Comment (QuLogic) [inline proj.4/build.sh]: Looks like it worked, so OSGeo/proj.4#317.

**Comment by pelson** [comment]
> @pelson does Obvious-CI creates the $VER variable for numpy? Is this an approach you want to pursue?

Yep, I'll take a look at upgrading obvious-ci for the latest conda-build. I will also update to use skip properly (see https://github.com/pelson/Obvious-CI/issues/32).

File: proj.4/build.sh (hunk 2 above)
Comment (ocefpaf) [inline proj.4/build.sh]: Yep that did the trick.  Thanks @QuLogic.

**Comment by pelson** [comment]
@ocefpaf - would you mind re-basing this?

**Comment by ocefpaf** [comment]
Done!

**Pull request merged**